packages/template-retail-react-app/app/hooks/use-active-data.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/*global dw*/
+import {ACTIVE_DATA_ENABLED} from '@salesforce/retail-react-app/app/constants'
+
+const useActiveData = () => {
+    return {
+        async sendViewProduct(category, product, type) {
+            if (!ACTIVE_DATA_ENABLED) {
+                return
+            }
+            try {
+                if (dw?.ac) {
```

Comment:
```suggestion
    const canTrack = () => ACTIVE_DATA_ENABLED && typeof dw !== 'undefined'
    return {
        async sendViewProduct(category, product, type) {
            if (!canTrack()) return
            try {
                if (dw.ac) {
```

As demonstrated on our call yesterday, the optional chaining operator is not a sufficient guard if the `dw` global has not been set. Only `typeof dw` guards against undeclared variables. I'd recommend implement a `canTrack` helper method, and using it as a guard at the top of each individual method (`sendViewProduct`, `sendViewSearch`, `sendViewCategory`, and `trackPage`).